### ISSUE TITLE
A question when reproducing the paper

### ISSUE THREAD

**Issue by xyz321123** [xiangyue9607/SanText#2]
Hello, when I use SanText.py to process the SST-2 dataset, I find that the tokens in it become wordpieces, what should I do with these wordpieces so that I can use this dataset in the mask token inference attack experiment? Can you give me the relevant code?
Thanks a lot.
![image](https://user-images.githubusercontent.com/49829553/158367634-4e718843-edc8-4fec-8149-54117920e065.png)

**Comment by xyz321123** [comment]
Sorry, just found out this is Bert's special token, 
But how do I use this special token in the Mask Token Inference Attack experiment, do I change the entire [unused*] to [MASK] and then predict it?

**Comment by xiangyue9607** [comment]
I uploaded our MASK token inference attack code. Please let me know if you have questions

**Comment by xyz321123** [comment]
Thank you.
And when I run the mask token inference attack code, I found the problem as shown in the picture.
![image](https://user-images.githubusercontent.com/49829553/173718456-50235781-3cc2-442e-98b8-689977b82cc5.png)
I debugged my code and found that len(batch[0])=len(batch[1])=len(batch[2])=256, but len(prediction)=277, I don't understand what's going on.
I found that I had prediction.shape=torch.Size([256, 64]) before running this one, but prediction.shape=torch.Size([277]) after running this one:
**_prediction = prediction[torch.where(batch[0]==103)]_**

**Comment by xyz321123** [comment]
Might SANTEXT based on Bert embeddings replace tokens with [MASK] token? I found that SANTEXT outputs sentences like below, which may be the cause of the error in the mask token inference code.
![image](https://user-images.githubusercontent.com/49829553/173778200-a51f9237-baf3-4eca-825e-137dcce7609d.png)
